src/Illuminate/Routing/ResourceRegistrar.php

```diff
@@ -313,6 +319,24 @@ protected function getShallowName($name, $options)
                     : $name;
     }
 
+    /**
+     * Set the route's binding fields if the resource is scoped.
+     *
+     * @param  \Illuminate\Routing\Route  $route
+     * @param  array  $bindingFields
+     * @return void
+     */
+    protected function setResourceBindingFields($route, $bindingFields)
```

Comment:
What is the purpose of this method? What does it do?

Reply:
It sets the binding fields of the route (if the `->scoped()` method was called) that will be used by the `ImplicitRouteBinding` class. By default all fields are null, so the default route key of the model will be used. Passing an array to the `scoped()` method will specify which keys should be used for the models.

It does the same as the `Route::get('api/users/{user}/posts/{post:slug}', ...)` but for each routes that are being registered by the resource registrar.